Source/Core/Core/HW/EXI/EXI_DeviceMemoryCard.cpp

```diff
@@ -236,8 +236,11 @@ void CEXIMemoryCard::SetupRawMemcard(u16 size_mb)
       SConfig::GetDirectoryForRegion(SConfig::ToGameCubeRegion(SConfig::GetInstance().m_region));
   MemoryCard::CheckPath(filename, region_dir, is_slot_a);
 
-  if (size_mb == Memcard::MBIT_SIZE_MEMORY_CARD_251)
-    filename.insert(filename.find_last_of('.'), ".251");
+  if (size_mb < Memcard::MBIT_SIZE_MEMORY_CARD_2043)
+  {
+    filename.insert(filename.find_last_of('.'),
+                    fmt::format(".{}", size_mb * Memcard::MBIT_TO_BLOCKS - Memcard::MC_FST_BLOCKS));
```

Comment:
Perhaps GCMemcard.h should have a helper function like:

```cpp
// Or whatever name is more appropriate
constexpr u32 ToBlockSize(u32 size_mb) {
  return (size_mb * Memcard::MBIT_TO_BLOCKS) - Memcard::MC_FST_BLOCKS;
}
```

Given the expression is primarily the same in several different places, with the only difference being that the size_mb part of the expression is different.

Reply:
Heh, I was just thinking exactly the same thing after I submitted the PR.

Reply:
How's that?